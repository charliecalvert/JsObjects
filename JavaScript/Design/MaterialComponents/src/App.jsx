import React, { Component } from 'react';
// import PropTypes from 'prop-types';
// import { withStyles } from '@mui/styles';
// import { styles } from './elf-styles';
import Grid from '@mui/material/Grid';
import Paper from '@mui/material/Paper';
import Typography from '@mui/material/Typography';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemText from '@mui/material/ListItemText';
import ListItemIcon from '@mui/material/ListItemIcon';
import StarIcon from '@mui/icons-material/Star';
import classNames from 'classnames';

class App extends Component {
    render() {
        // const { classes } = this.props;

        /*
         * We want to be dry, so we create this function to avoid
         * repeating the same code over and over in JSX.
         */
        const getListItem = (text, id) => {
            return (
                <ListItem id={id}>
                    <ListItemIcon>
                        <StarIcon />
                    </ListItemIcon>
                    <ListItemText>{text}</ListItemText>
                </ListItem>
            );
        };

        /*
         * We want to combine elfTypography with elfParagraph. React
         * doesn't support this out of the box, so we use the very
         * popular classNames library. We use it below in Section Two.
         */
 /*        const elfParagraph = classNames(
            classes.elfTypography,
            classes.elfParagraph
        ); */

        return (
            <React.Fragment>
                <div>
                    <Grid container spacing={24}>
                        <Grid item xs={12}>
                            {/*This empty Grid puts space before the first section*/}
                        </Grid>

                        <Grid item xs={12}>
                            <Paper>
                                <Typography variant="h4">
                                    Section One
                                </Typography>
                            </Paper>
                        </Grid>

                        <Grid item xs={12}>
                            <Paper>
                                <Typography variant="h6">
                                    Section Two
                                </Typography>
                                <Typography
                                    // className={elfParagraph}
                                    align="left"
                                    variant="body1"
                                >
                                    In <b>src/index.js</b> we create a theme and
                                    wrap the entire app in the theme. We also
                                    use <b>CssBaseline</b>.
                                </Typography>

                                <Typography
                                    //
                                    align="left"
                                    variant="body1"
                                >
                                    View the source of this component to see how
                                    we wrap our content in the following
                                    elements:
                                </Typography>
                                <List>
                                    {getListItem('layout', 'lo')}
                                    {getListItem(
                                        'A Grid with container and spacing of 24',
                                        'gridContainerSpace24'
                                    )}
                                </List>
                                <Typography
                                    //
                                    align="left"
                                    variant="body1"
                                >
                                    Inside the four items we create one or more
                                    sections for our document. Each section has
                                    this structure:
                                </Typography>
                                <List>
                                    {getListItem(
                                        'A Grid item with xs = 12',
                                        'gridItem12'
                                    )}
                                    {getListItem(
                                        'Paper with a className of paperLion',
                                        'paperLion'
                                    )}
                                    {getListItem(
                                        'And then some content',
                                        'content'
                                    )}
                                </List>
                            </Paper>
                        </Grid>
                    </Grid>
                </div>
            </React.Fragment>
        );
    }
}

/* App.propTypes = {
    classes: PropTypes.object.isRequired
}; */

export default App;
