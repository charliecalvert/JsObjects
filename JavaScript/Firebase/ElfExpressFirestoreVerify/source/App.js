import React, { Component } from 'react';
import { getFirebaseToken, makeParams } from './get-firebase-token';

class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            error: 'none',
            result: 'unknown',
            name: 'unknown',
            email: 'unknown',
            photoUrl: 'unknown',
            stack: 'none',
            items: [{ id: -4, data: 'none' }]
        };
    }

    setData = (result) => {
        console.log('FETCH RESULT', result);
        this.setState(result);
    };

    runQuery = event => {
        const url = event.currentTarget.dataset.url;
        getFirebaseToken(this.setData)
            .then(response => {
                fetch(url + makeParams({ token: response.token }))
                    .then(response => {
                        return response.json();
                    })
                    .then(result => {
                        this.setData(result);
                    })
                    .catch(ex => {
                        console.log(ex);
                        this.setState({ error: ex.message, stack: ex.stack });
                    });
            })
            .catch(ex => {
                console.log(ex);
            });
    };

    runBatchRead = event => {
        const url = event.currentTarget.dataset.url;
        getFirebaseToken(this.setData)
            .then(response => {
                fetch(url + makeParams({ token: response.token }))
                    .then(response => {
                        return response.json();
                    })
                    .then(result => {
                        this.setData({ items: result });
                    })
                    .catch(ex => {
                        console.log(ex);
                        this.setState({ error: ex.message, stack: ex.stack });
                    });
            })
            .catch(ex => {
                console.log(ex);
            });
    };

    render() {
        return (
            <div>
                <h1>Elf Express Firestore Verify</h1>
                <button onClick={() => window.open('elf-sign-in.html')}>
                    Login
                </button>
                <a href="/worker?title=FirebaseLogout">Logout</a>

                <h2>Batch and Snapshot</h2>
                <button onClick={this.runQuery} data-url="/write-batch">
                    Test Write Batch
                </button>
                <button onClick={this.runBatchRead} data-url="/read-snapshot">
                    Test Read Snapshot
                </button>

                <h2>Writes</h2>
                <p>
                    <b>Result</b>: {this.state.result}
                </p>

                <h2>Reads</h2>
                <p>
                    <b>ID</b>: {this.state.items[0].id}
                </p>
                <p>
                    <b>Data</b>: {this.state.items[0].data}
                </p>

                <h2>Errors</h2>
                <p>
                    <b>Status Message</b>: {this.state.error}
                </p>
                <p>
                    <b>Stack</b>: {this.state.stack}
                </p>
            </div>
        );
    }
}

export default App;
