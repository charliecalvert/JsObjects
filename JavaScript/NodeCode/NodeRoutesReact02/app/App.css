.app {
  background-color: green;
  height: 50px;
  border: 1px solid blue;
}

.bap {
  background-color: red;
  height: 25px;
}

body {
    background-color: #EEE;
}

h1, h2 {
    text-shadow: 5px 5px 5px #AAA;
}

.neon {
    color: #D0F8FF;
    text-shadow: 0 0 5px #A5F1FF, 0 0 10px #A5F1FF,
    0 0 20px #A5F1FF, 0 0 30px #A5F1FF,
    0 0 40px #A5F1FF;
}

/**
 * Shadow
 */
a.button::before {
    -webkit-border-radius: 3px;
    -moz-border-radius: 3px;
    -webkit-box-shadow: #959595 0 2px 5px;
    -moz-box-shadow: #959595 0 2px 5px;
    border-radius: 3px;
    box-shadow: #959595 0 2px 5px;
    content: "";
    display: block;
    height: 100%;
    left: 0;
    padding: 2px 0 0;
    position: absolute;
    top: 0;
    width: 100%; }

a.button:active::before { padding: 1px 0 0; }

/**
 * Grey
 * Thanks to https://www.sanwebe.com/
 */
a.button {
    -moz-box-shadow: inset 0 0 0 1px #63ad0d;
    -webkit-box-shadow: inset 0 0 0 1px #63ad0d;
    -moz-border-radius: 3px;
    -webkit-border-radius: 3px;
    background: #eee;
    background: -webkit-gradient(linear, 0 0, 0 bottom, from(#eee), to(#e2e2e2));
    background: -moz-linear-gradient(#eee, #e2e2e2);
    background: linear-gradient(#eee, #e2e2e2);
    border: solid 1px #d0d0d0;
    border-bottom: solid 3px #b2b1b1;
    border-radius: 3px;
    box-shadow: inset 0 0 0 1px #f5f5f5;
    color: #555;
    display: inline-block;
    font: bold 12px Arial, Helvetica, Clean, sans-serif;
    margin: 0 25px 25px 0;
    padding: 10px 20px;
    position: relative;
    text-align: center;
    text-decoration: none;
    text-shadow: 0 1px 0 #fafafa; }

a.button:hover {
    background: #e4e4e4;
    background: -webkit-gradient(linear, 0 0, 0 bottom, from(#e4e4e4), to(#ededed));
    background: -moz-linear-gradient(#e4e4e4, #ededed);
    background: linear-gradient(#e4e4e4, #ededed);
    border: solid 1px #c2c2c2;
    border-bottom: solid 3px #b2b1b1;
    box-shadow: inset 0 0 0 1px #efefef; }

a.button:active {
    background: #dfdfdf;
    background: -webkit-gradient(linear, 0 0, 0 bottom, from(#dfdfdf), to(#e3e3e3));
    background: -moz-linear-gradient(#dfdfdf, #e3e3e3);
    background: linear-gradient(#dfdfdf, #e3e3e3);
    border: solid 1px #959595;
    box-shadow: inset 0 10px 15px 0 #c4c4c4;
    top:2px;}

/**
 * elf - Special Edition
 */
a.button.elf {
    background: #99b6df;
    background: -webkit-gradient(linear, 0 0, 0 bottom, from(#99b6df), to(#638ec8));
    background: -moz-linear-gradient(#99b6df, #638ec8);
    background: linear-gradient(#99b6df, #638ec8);
    border: solid 1px #6d94ce;
    border-bottom: solid 3px #3867ac;
    box-shadow: inset 0 0 0 1px #bbcfeb;
    color: #fff;
    text-shadow: 0 1px 0 #3c61ab; }

a.button.elf:hover {
    background: #638ec8;
    background: -webkit-gradient(linear, 0 0, 0 bottom, from(#638ec8), to(#99b6df));
    background: -moz-linear-gradient(#638ec8, #99b6df);
    background: linear-gradient(#638ec8, #99b6df);
    border: solid 1px #6d94ce;
    border-bottom: solid 3px #3867ac;
    box-shadow: inset 0 0 0 1px #bbcfeb; }

a.button.elf:active {
    background: #638ec8;
    background: -webkit-gradient(linear, 0 0, 0 bottom, from(#638ec8), to(#99b6df));
    background: -moz-linear-gradient(#638ec8, #99b6df);
    background: linear-gradient(#638ec8, #99b6df);
    border: solid 1px #6d94ce;
    box-shadow: inset 0 10px 15px 0 #4176c4; }