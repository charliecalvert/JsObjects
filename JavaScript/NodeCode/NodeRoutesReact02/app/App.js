import React from 'react';
import styles from './App.css';

export default class App extends React.Component {
    render() {
        return (
            <div className={styles.app}>
                <h1>Node Routes React 02</h1>
                <p>
                    A simple Node Express program. This one differs from Version
                    01 in that it breaks the code up into different folders:
                </p>

                <ul>
                    <li>Public: Client side code</li>
                    <li>Source: Server side business logic.</li>
                    <li>
                        Routes: The place where the server handles URLs (routes)
                        sent from client
                    </li>
                    <li>
                        bin: A stub for starting the application. Start it with
                        one of these options:
                        <ul>
                            <li>node bin/www</li>
                            <li>npm start</li>
                        </ul>
                    </li>
                </ul>

                <hr />
            </div>
        );
    }
}
