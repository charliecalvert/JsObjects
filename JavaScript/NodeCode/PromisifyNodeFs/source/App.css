#fileNames {
    font-family: Arial, Helvetica, sans-serif;
    font-size: large;
    border-collapse: collapse;
    width: 100%;
}

#fileNames td, #fileNames th {
    border: 1px solid #ddd;
    padding: 8px;
}

#fileNames tr:nth-child(even){background-color: #f2f2f2;}

#fileNames tr:hover {background-color: #ddd;}

#fileNames th {
    padding-top: 12px;
    padding-bottom: 12px;
    text-align: left;
    background-color: #1177ff;
    color: white;
}

p, li {
    font-size: large;
}

button {
    margin-top: 25px;
    background-color: #1177ff;
    border: none;
    border-radius: 8px;
    color: white;
    padding: 7px 12px;
    text-align: center;
    display: inline-block;
    font-size: large;
}
