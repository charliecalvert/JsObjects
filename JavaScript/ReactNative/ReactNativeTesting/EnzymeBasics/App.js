import React from 'react';
import { Text, View, Button } from 'react-native';
import Address from './Address';
import elfStyles from './elf-styles';

export default class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            test: 'unknown',
            result: 'unknown',
            ex: 'ff'
        };
        this.fetchMicro = this.fetchMicro.bind(this);
    }

    clickTest = () => {
        this.setState({ test: 'foo is known' });
    };

    fetchMicro() {
        const that = this;
        const ips = ['ccalvert.com', '168.156.43.39'];
        return fetch('http://' + ips[0] + ':30027/you-rang')
            .then(response => response.json())
            .then(function(result) {
                console.log(result);
                that.setState({
                    result: result.result
                });
            })
            .catch(function() {
                that.setState({ foo: 'qux error' });
            });
    }

    render() {
        return (
            <View style={elfStyles.container}>
                <Text id="testClosed">Test: {this.state.test}</Text>
                <Text>You Rang: {this.state.result}</Text>
                <Button id="clickTest" title="Test" onPress={this.clickTest}>
                    Foobar
                </Button>
                <Button
                    id="fetchMicro"
                    title="Fetch Micro"
                    onPress={this.fetchMicro}
                >
                    Foobar
                </Button>
                <Address />
            </View>
        );
    }
}
