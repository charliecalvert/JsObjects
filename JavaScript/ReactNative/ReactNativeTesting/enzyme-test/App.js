import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default function App() {
  return (
    <View style={styles.container}>
      <h1>You picked foo</h1>
      <Text>Open up App.js to start working on your app!</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
