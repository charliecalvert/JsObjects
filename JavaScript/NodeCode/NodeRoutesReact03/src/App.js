import React, { Component } from 'react';
import './App.css';

class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            title: 'waiting for you to click...',
            query: [],
            queryBar: 'waiting for you to click...',
            queryCount: 'waiting for you to click...',
            params: 'waiting for you to click...'
        };
    }

    callFoo = () => {
        const that = this;
        fetch('/foo?bar=qux&count=5')
            .then(function(response) {
                return response.json();
            })
            .then(function(json) {
                console.log('parsed json', json);
                const query = Object.keys(json.query).map(function(key) {
                    return [key, ' ', json.query[key], ' '];
                });
                that.setState({
                    title: json.title,
                    query: query,
                    queryBar: json.query.bar,
                    queryCount: json.query.count,
                    params: JSON.stringify(json.params),
                });
            })
            .catch(function(ex) {
                console.log(
                    'parsing failed, URL bad, network down, or similar',
                    ex
                );
            });
    };

    render() {
        return (
            <div className="App">
                <header>
                    <h1>Simple React and Express</h1>
                    <p>Title: {this.state.title}</p>
                    <p>Query: {this.state.query}</p>
                    <p>Query Bar: {this.state.queryBar}</p>
                    <p>Query Count: {this.state.queryCount}</p>
                    <p>Params: {this.state.params}</p>
                </header>
                <main>
                    <button onClick={this.callFoo}>Call Foo</button>
                </main>
                <footer>
                    <p>&copy; by Charlie Calvert</p>
                </footer>
            </div>
        );
    }
}

export default App;
