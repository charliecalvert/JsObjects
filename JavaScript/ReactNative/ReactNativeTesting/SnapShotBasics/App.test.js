import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { StyleSheet, Text, Component } from 'react-native';
configure({ adapter: new Adapter() });

describe('App test', function() {
    it('renders without crashing', () => {
        const rendered = renderer.create(<App />).toJSON();
        expect(rendered).toBeTruthy();
    });

    it('test default output of App', () => {
        const wrapper = shallow(<App />);
        expect(wrapper).toMatchSnapshot();
    });

    it('test specific node with property/attribute called bar', () => {
        const wrapper = shallow(<App />);
        console.log(wrapper.debug());
        const foo = <Text bar="3">Open</Text>;
        expect(wrapper.contains(foo)).toBe(true);
    });

    it('test for match element without props (ignores bar)', () => {
        const wrapper = shallow(<App />);
        console.log(wrapper.debug());
        const openText = <Text>Open</Text>;
        expect(wrapper.containsMatchingElement(openText)).toEqual(true);
    });
});
