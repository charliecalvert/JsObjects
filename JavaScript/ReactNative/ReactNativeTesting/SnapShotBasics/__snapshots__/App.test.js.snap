// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test test 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    bar="3"
    ellipsizeMode="tail"
  >
    Open
  </Text>
</Component>
`;
